src/misc/base16/base16_encode.c

```diff
@@ -49,12 +49,12 @@ int base16_encode(const unsigned char *in,  unsigned long  inlen,
       *outlen = x;
       return CRYPT_BUFFER_OVERFLOW;
    }
-   *outlen = x;
+   x--;
```

Comment:
that's exactly why I went with the way it was... `outlen` having two different return values for the same input...